tuf/data/roles.go

```diff
@@ -176,3 +189,74 @@ func (r Role) IsDelegation() bool {
 	targetsBase := fmt.Sprintf("%s/", ValidRoles[CanonicalTargetsRole])
 	return strings.HasPrefix(r.Name, targetsBase)
 }
+
+// AddKeys merges the ids into the current list of role key ids
+func (r *Role) AddKeys(ids []string) {
+	r.KeyIDs = mergeStrSlices(r.KeyIDs, ids)
+}
+
+// AddPaths merges the paths into the current list of role paths
+func (r *Role) AddPaths(paths []string) error {
+	if len(paths) == 0 {
+		return nil
+	}
+	if len(r.PathHashPrefixes) > 0 {
+		return ErrInvalidRole{Role: r.Name, Reason: "attempted to add paths to role that already has hash prefixes"}
+	}
+	r.Paths = mergeStrSlices(r.Paths, paths)
+	return nil
+}
+
+// AddPathHashPrefixes merges the prefixes into the list of role path hash prefixes
+func (r *Role) AddPathHashPrefixes(prefixes []string) error {
+	if len(prefixes) == 0 {
+		return nil
+	}
+	if len(r.Paths) > 0 {
+		return ErrInvalidRole{Role: r.Name, Reason: "attempted to add hash prefixes to role that already has paths"}
+	}
+	r.PathHashPrefixes = mergeStrSlices(r.PathHashPrefixes, prefixes)
+	return nil
+}
+
+// RemoveKeys removes the ids from the current list of key ids
+func (r *Role) RemoveKeys(ids []string) {
+	r.KeyIDs = subtractStrSlices(r.KeyIDs, ids)
+}
+
+// RemovePaths removes the paths from the current list of role paths
+func (r *Role) RemovePaths(paths []string) {
```

Comment:
We should document it somewhere. Although we're unlikely to expose path
hash prefixes any time soon. They come with some significant caveats that
in the best case are confusing, and in the worst, create exploits.